hash/doc.go

```diff
@@ -1,5 +1,15 @@
 // Package hash provides MiMC hash function defined over implemented curves
 //
+// # Importing hash functions
+//
+// The package follows registration pattern for importing hash functions. To
+// import all known hash functions in gnark-crypto, import the
+// [github.com/consensys/gnark-crypto/hash/all] package in your code. To import
+// only a specific hash, then import the corresponding package directly, e.g.
+// [github.com/consensys/gnark-crypto/ecc/bn254/fr/mimc]. The import format should be:
+//
+//	import _ "github.com/consensys/gnark-crypto/ecc/bn254/fr/mimc"
```

Comment:
so, package doc thing, do I understand correctly that the recommended way is to use directly the New function in the "curve-typed" import path and NOT use hash/ anymore ?

But that we keep hash/ for retro compat reasons? 

Can we make that explicit in the doc?